dataprofiler/profilers/histogram_utils.py

```diff
@@ -14,6 +14,293 @@
 )
 
 
+def _ptp(maximum, minimum):
+    """Peak-to-peak using minimum and maximum value of a dataset.
+
+    This implementation avoids the problem of signed integer arrays having a
+    peak-to-peak value that cannot be represented with the array's data type.
+    This function returns an unsigned value for signed integer arrays.
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+    """
+    return np.subtract(maximum, minimum)
+
+
+def _calc_doane_bin_width_from_profile(profile):
+    """
+    Doane's histogram bin estimator reworked to use profiles.
+
+    Improved version of Sturges' formula which works better for
+    non-normal data. See
+    stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    Parameters
```

Comment:
I'm thinking we should reformat to use our format for docstrings until we want to convert to a different technique.

Reply:
I am not sure. this may depend on Justin's feedback so Ill wait it out for now